server/src/main/java/org/uiautomation/ios/instruments/InstrumentsCommandLine.java

```diff
@@ -136,8 +136,14 @@ public void start(long timeout) throws InstrumentsFailedToStartException {
       log.fine("registration request received" + session.getCachedCapabilityResponse());
       if (!success) {
         log.warning("instruments crashed (" + ((System.currentTimeMillis() - waitStartTime) / 1000) + " sec)".toUpperCase());
-        throw new InstrumentsFailedToStartException(
-            "Didn't get the capability back.Most likely, instruments crashed at startup.");
+        instruments.waitFor(-1);
```

Comment:
This is infinite wait, a blocking call. Since waitForUIScriptToBeStarted failed, it is possible that the script took too much time to start or is hung. In that case this would wait for eternity. Wait with a timeout instead and on timeout throw InstrumentsFailedToStartException